leader.go

```diff
@@ -106,7 +107,7 @@ func nodeLists(nodes []*api.Node, insts []*api.ServiceEntry,
 func (a *Agent) commitOps(ops api.KVTxnOps) bool {
 	success, results, _, err := a.client.KV().Txn(ops, nil)
 	if err != nil || !success {
-		a.logger.Printf("[ERR] Error writing state to KV store: %v, %v", err, results)
+		a.logger.Error("Error writing state to KV store", "error", hclog.Fmt("%v, %v", err, results))
```

Comment:
Can you split "error" k/v pair into something like `"results", results, "error", err`?